test/boulder-tools/hotfix_release.sh

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env bash
+
+# -e Stops execution in the instance of a command or pipeline error.
+# -u Treat unset variables as an error and exit immediately.
+set -eu
+
+STATUS="FAILURE"
+RST=$(tput sgr0)
+RED=$(tput bold && tput setaf 1)
+GRN=$(tput bold && tput setaf 2)
+BLU=$(tput bold && tput setaf 4)
+
+function exit_msg() {
+  # complain to STDERR and exit with error
+  echo "${*}" >&2
+  exit 2
+}
+
+function print_outcome() {
+  if [ "${STATUS}" == SUCCESS ]
+  then
+    echo
+    echo "${GRN}${STATUS}${RST}"
+  else
+    echo
+    echo "${RED}${STATUS}${RST}"
+  fi
+}
+
+function print_heading() {
+  echo
+  echo "${BLU}${1}${RST}"
+}
+
+function not_yes_no() {
+    case "${1}" in
+      [yY][eE][sS]|[yY]) create_branch="yes" && return 1 ;;
+      [nN][oO]|[nN]) create_branch="no" && return 1 ;;
+      *) return 0 ;;
+    esac
+}
+
+function get_user_input() {
+    while [ -z "${create_branch}" ]; do
+        read -p "${1}" create_branch
+        if not_yes_no "${create_branch}" 
+        then
+            echo "Need [yes/no], got [${create_branch}]"
+            create_branch=""
+        fi
+    done
+}
+
+# On EXIT, trap and print outcome.
+trap "print_outcome" EXIT
+
+print_heading "Ensuring main branch is up to date"
+git fetch --all
+git checkout main
+git pull --rebase
+
+print_heading "Fetching details for the most recent release tag"
+latest_tag_sha=$(git show-ref --tags | tail -1 | awk '{print $1}')
+latest_tag_name=$(git show-ref --tags | tail -1 | awk '{print $2}' | sed 's|refs\/tags\/||')
+
+print_heading "Latest tag:"
+echo "${latest_tag_name}"
+echo "${latest_tag_sha}"
+
+
+print_heading "Hotfix release tag:"
+if [[ "${latest_tag_name: -1}" =~ ^[0-9] ]]
+then
+    new_tag_name="${latest_tag_name}a"
+else
+    tag_with_last_char_removed=$(echo "${latest_tag_name}" | sed 's|.$||')
+    next_tag_letter=$(echo "${latest_tag_name: -1}" |  tr "0-9a-z" "1-9a-z_")
+    new_tag_name="${tag_with_last_char_removed}${next_tag_letter}"   
+fi
+echo "${new_tag_name}"
+
+create_branch=""
+get_user_input "Create hotfix release branch: ${new_tag_name}? "
+if [ "${create_branch}" = yes ]
+then
+    print_heading "Creating new branch ${new_tag_name} @ ${latest_tag_sha}"
```

Comment:
Don't create a branch with the same name as the new tag: create a branch with the same name as the _base release tag_, or re-use one such already-created branch if we're creating a `...b` or subsequent release tag.

You can see an example of the structure here: https://github.com/letsencrypt/boulder/tree/release-branch-2021-08-31
This branch is named `release-branch-2021-08-31` (because it is based off of the commit with that tag), and then two more commits unique to that branch are also tagged, as `release-branch-2021-08-31a` and `release-branch-2021-08-31b`.

This will also make a lot of the following code much clearer imo, as you won't be using `new_tag_name` interchangeably sometimes as a tag and sometimes as a branch.